Fix acl

fabbione opened the pull request.

fabbione committed:
[acl] make knet_link_add_acl a wrapper to knet_link_insert_acl

remove lots of duplicated code

Signed-off-by: Fabio M. Di Nitto <fdinitto@redhat.com>

fabbione committed:
[acl] major internal rework

ACL internal management was somewhat broken and didn´t really
do what was advertised for dynamic links.

- fix ACL to be truely per host/link as public API advertise
- move ACL checking when host/link are known
- add extensive tests
- fix various minor bugs that went undetected for a bit
- enable access list testings on more tests

Signed-off-by: Christine Caulfield <ccaulfie@redhat.com>
Signed-off-by: Fabio M. Di Nitto <fdinitto@redhat.com>

fabbione commented:
retest this please

fabbione commented:
@chrissie-c commits have been squashed.
There are still some issues in CI with memcheck.
https://ci.kronosnet.org/view/knet/job/knet-build-all-nonvoting-memcheck/37/

Some jobs failed with timeout of 240 minutes, I increased that to 360. Others have failed in some weird ways (centos 8 for example).


20:30:07 [knet: 0x67de040]: [debug] handle: Links access lists are disabled
20:30:07 [knet: 0x67de040]: [debug] rx: Source host 2 not reachable yet. Discarding packet.
20:30:07 [knet]: No logs in the last 60 seconds
20:30:07 [knet]: No logs in the last 60 seconds

the memcheck dump on centos.-8 is a result of not freeing resources and can be ignored.

fabbione commented:
@chrissie-c did a re-run with higher timeouts. rawhide and powerpc are failing.

chrissie-c commented:
I don't know whether I'm doing something wrong, but fun_pmtud_crypto_test takes around 8 hours to run on anything I have here (not even under valgrind!)

fabbione commented:
> I don't know whether I'm doing something wrong, but fun_pmtud_crypto_test takes around 8 hours to run on anything I have here (not even under valgrind!)

that specific test is not executed automatically and it's not part of make check or make check-memcheck.

fabbione committed:
[transports] remove unused internal ACL API

with the new ACL code, this internal API is no longer used

Signed-off-by: Fabio M. Di Nitto <fdinitto@redhat.com>

fabbione committed:
[tests] make execution of functional testing optionals

Signed-off-by: Fabio M. Di Nitto <fdinitto@redhat.com>

fabbione commented:
@wferi I know buster is in super strict RC mode now. 2 or 3 of the commits here are rather important, so raising awareness for you guys. The first SCTP fix (fix invalid memory access) is pretty self explanatory. The [acl] major internal work is also important as it fixes acl behavior. corosync usecase is not affected, but other applications might. The 3rd one is the [sctp] remove sockets from epoll earlier. This is important only for slow arches as it can cause knet to become less responsive when a SCTP socket is disconnected.
We are in the process of heavily testing all those changes, but please keep an eye as once they are fully stable they will hit the next stable release.

fabbione commented:
retest this please

fabbione commented:
retest this please

chrissie-c committed:
send: Fix send_sync()

send_sync needs a filter to work - enforce this
Clear out some arrays that could have random values

fabbione commented:
retest this please

fabbione committed:
[tests] fix fun_onwire_upgrade checks

- handle wait_for_nodes_states properly (detected by coverity scan)
- wait for the correct number of nodes

Signed-off-by: Fabio M. Di Nitto <fdinitto@redhat.com>

fabbione commented:
retest this please

chrissie-c committed:
test: Use a pipe to synchronise wait_for_xxx events

The condition variable thing clearly not working for us.

fabbione commented:
retest this please

fabbione commented:
retest this please

chrissie-c reviewed (approved):
Pre-emptive ACK
